modules/Microsoft.HealthDataServices/deploy.bicep

```diff
@@ -0,0 +1,58 @@
+@description('Required. The name of the Health Data Services Workspace service.')
+@maxLength(50)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled.')
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+// =========== //
+// Deployments //
+// =========== //
+
+resource health 'Microsoft.HealthcareApis/workspaces@2022-06-01' = {
```

Comment:
This template should be put in its own subfolder `workspaces` - and all other 'nested/child' resources should be nested folders in it. That means, the structure of the module should look as follows:
- Microsoft.HealthcareApis
  - workspaces
    - .test
    - deploy.bicep
    - readme.md
    - version.json
    - dicomservices
      - deploy.bicep
      - readme.md
      - version.json 
    - fhirservices
      - deploy.bicep
      - readme.md
      - version.json 

Then, the parent module 'workspaces' should reference its children - essentially enabling you to deploy the children via their parent and standalone.

For have further guidelines [here](https://github.com/Azure/ResourceModules/wiki/The%20library%20-%20Module%20design). 

You already nailed most of that stuff, so it's more about some structural changes rather than content changes.

Reply:
OK yes I will work on this restructure.

Reply:
Updated structure but I have not worked on the test files